query/sources/common/src/main/java/org/codice/ddf/admin/sources/fields/type/SourceConfigUnionField.java

```diff
@@ -83,16 +92,79 @@ public SourceConfigUnionField credentials(String username, String password) {
         return this;
     }
 
+    public SourceConfigUnionField address(String hostname, int port) {
+        address.hostname(hostname);
+        address.port(port);
+        return this;
+    }
+
+    // Getters
+    public String sourceName() {
+        return sourceName.getValue();
+    }
+
+    public StringField sourceNameField() {
+        return sourceName;
+    }
+
+    public CredentialsField credentials() {
+        return this.creds;
+    }
+
+    public String factoryPid() {
+        return factoryPid.getValue();
+    }
+
+    public FactoryPid factoryPidField() {
+        return factoryPid;
+    }
+
+    public String servicePid() {
+        return servicePid.getValue();
+    }
+
+    public ServicePid servicePidField() {
+        return servicePid;
+    }
+
+    public AddressField address() {
+        return address;
+    }
+
+    public String endpointUrl() {
+        return endpointUrl.getValue();
+    }
+
+    public UrlField endpointUrlField() {
+        return endpointUrl;
+    }
+
     @Override
     public List<Field> getFields() {
-        return ImmutableList.of(pid, sourceName, endpointUrl, creds);
+        return ImmutableList.of(factoryPid, servicePid, sourceName, endpointUrl, creds, address);
     }
 
     @Override
     public void initializeFields() {
-        pid = new PidField();
-        sourceName = new StringField("sourceName");
-        endpointUrl = new UrlField("endpointUrl");
+        factoryPid = new FactoryPid();
+        servicePid = new ServicePid();
+        sourceName = new StringField(SOURCE_NAME_FIELD);
```

Comment:
I don't see how this would be different than a StringField. If we put that the source name must be unique in a SourceName field, we are assuming that it must be unique for anyone using it, which may not be the case. The uniqueness is enforced by the actions instead.